components/builder-http-gateway/src/http/middleware.rs

```diff
@@ -209,8 +212,11 @@ impl AfterMiddleware for Cors {
     }
 }
 
-pub fn session_create_github(req: &mut Request, token: String) -> IronResult<Session> {
-    let github = req.get::<persistent::Read<GitHubCli>>().unwrap();
+pub fn session_create_github(
+    github: &GitHubClient,
```

Comment:
I wouldn't change the signature here - you can get a ref of GitHub from the req. We're building our middleware in a consistent way across all http-gateways so these functions should always have access to GitHub/RouteClient

Reply:
Hmm, I did try that and the unwrap panic'd

Reply:
That'd be weird, we would be calling this out of order then before the GitHub creds are available